src/sc/sidechaintypes.h

```diff
@@ -29,96 +61,104 @@ struct CFieldPtrDeleter
 };
 typedef std::shared_ptr<field_t> wrappedFieldPtr;
 
-class CFieldElement
+class CFieldElement : public CZendooCctpObject
 {
 public:
-    CFieldElement();
+    CFieldElement() = default;
     ~CFieldElement() = default;
-
     explicit CFieldElement(const std::vector<unsigned char>& byteArrayIn);
-    explicit CFieldElement(const uint256& value);
-    void SetByteArray(const std::vector<unsigned char>& byteArrayIn);
+    void SetByteArray(const std::vector<unsigned char>& byteArrayIn) override final;
 
-    CFieldElement(const CFieldElement& rhs) = default;
```

Comment:
We need a custom copy ctor for having the internal data (protected byteVector) copied as well. This also applies to other subclasses when a deep copy is required.